src/main/java/bejeweled/board/CombinationIterator.java

```diff
@@ -0,0 +1,35 @@
+package bejeweled.board;
+
+import java.util.ArrayList;
+
+public class CombinationIterator implements Iterator{
+	Combination combi;
+	int position = 0;
+	ArrayList<Gem> g;
+	
+	public CombinationIterator(Combination combi){
+		this.combi=combi;
+		g = combi.getGems();
+	}
+	
+	public Object next(){
```

Comment:
Could return a gem here instead of an object, saves you the cast to a gem when you call it.